app/controllers/ncr/work_orders_controller.rb

```diff
@@ -21,18 +21,32 @@ def edit
     end
 
     def update
+      @model_instance.assign_attributes(permitted_params)
       @model_instance.modifier = current_user
 
-      super
+      if errors.empty?
+        update_or_notify_of_no_changes
+        redirect_to proposal_path(@model_instance.proposal)
+      else
+        flash[:error] = errors
+        render :edit
+      end
+    end
 
-      if @model_changing
+    protected
+
+    def update_or_notify_of_no_changes
+      if attribute_changes?
+        ProposalUpdateRecorder.new(@model_instance).run
+        @model_instance.save
         @model_instance.setup_approvals_and_observers
         @model_instance.email_approvers
+        flash[:success] = "Successfully modified!"
+      else
+        flash[:error] = "No changes were made to the request"
       end
     end
```

Comment:
What I am finding as I make various updates is that there is a lot of different functionality required in the individual client_data controllers vs `UseCaseController` 

eg: here, we want to call `ProposalUpdateRecorder.new(@model_instance).run` only if there are no errors and there are changes for an `Ncr::WorkOrder`...so I am not sure how to do that for this controller only while sharing functionality with the `update` method in `UseCaseController`...suggestions?